aztec_macros/src/lib.rs

```diff
@@ -472,20 +485,30 @@ fn collect_crate_structs(crate_id: &CrateId, context: &HirContext) -> Vec<Struct
 }
 
 /// Substitutes the signature literal that was introduced in the selector method previously with the actual signature.
-fn transform_event(struct_id: StructId, interner: &mut NodeInterner) {
+fn transform_event(
+    struct_id: StructId,
+    interner: &mut NodeInterner,
+) -> Result<(), (AztecMacroError, FileId)> {
     let struct_type = interner.get_struct(struct_id);
     let selector_id = interner
-        .lookup_method(&Type::Struct(struct_type, vec![]), struct_id, "selector", false)
-        .expect("Selector method not found");
+        .lookup_method(&Type::Struct(struct_type.clone(), vec![]), struct_id, "selector", false)
+        .ok_or_else(|| {
+            let error = AztecMacroError::EventError {
+                span: struct_type.borrow().location.span,
+                message: "Selector method not found".to_owned(),
+            };
+            (error, struct_type.borrow().location.file)
+        })?;
     let selector_function = interner.function(&selector_id);
 
-    let compute_selector_statement = interner.statement(
-        selector_function
-            .block(interner)
-            .statements()
-            .first()
-            .expect("Compute selector statement not found"),
-    );
+    let compute_selector_statement =
+        interner.statement(selector_function.block(interner).statements().first().ok_or((
+            AztecMacroError::EventError {
+                span: struct_type.borrow().location.span,
+                message: "Compute selector statement not found".to_owned(),
+            },
+            struct_type.borrow().location.file,
+        ))?);
```

Comment:
Missed an `ok_or_else` here:
```suggestion
    let compute_selector_statement =
        interner.statement(selector_function.block(interner).statements().first().ok_or_else(|| {
            let error = AztecMacroError::EventError {
                span: struct_type.borrow().location.span,
                message: "Compute selector statement not found".to_owned(),
            };
            (error, struct_type.borrow().location.file)
        })?);
```